Adding sparse file SPC and SC templates in k8s-cstorpool and k8s-stor…

gprasath opened the pull request:
…ageclass roles respectively

Signed-off-by: gprasath <giridhara.prasad@cloudbyte.com>

**What this PR does / why we need it**:
- Added a template with sparse file SPC in k8s-cStorpool role.
- Creating cstorpool on sparse file if the extra-var passed is "sparsepool_create" and deleing it if the extra-var provided is "sparsepool_delete"
- Added a template with sparse file storage class spec in k8s-custom-storageclass role.
- Creating the storage class with sparse file specs. Deleting the copied templates at the end of play.

ksatchit commented:
@gprasath , can you please provide a sample output of the role execution with sparse file actions

gprasath commented:
Storage-class creation sample output:
```
PLAY [kubernetes-kubemasters] *************************************************************************************************************************************************************************************************************

TASK [Gathering Facts] ********************************************************************************************************************************************************************************************************************
task path: /home/giridhara_prasad/openebs/e2e/ansible/setup-openebs.yml:6
ok: [kubemaster01]
META: ran handlers

TASK [k8s-custom-storageclass : Get $HOME of K8s master for kubernetes user] **************************************************************************************************************************************************************
task path: /home/giridhara_prasad/openebs/e2e/ansible/roles/k8s-custom-storageclass/tasks/main.yml:2
changed: [kubemaster01 -> None] => {"changed": true, "cmd": "source ~/.profile; echo $HOME", "delta": "0:00:00.003868", "end": "2018-08-22 07:40:34.457972", "rc": 0, "start": "2018-08-22 07:40:34.454104", "stderr": "", "stderr_lines": [], "stdout": "/home/giridhara_prasad", "stdout_lines": ["/home/giridhara_prasad"]}

TASK [k8s-custom-storageclass : template] *************************************************************************************************************************************************************************************************
task path: /home/giridhara_prasad/openebs/e2e/ansible/roles/k8s-custom-storageclass/tasks/main.yml:10
changed: [kubemaster01] => {"changed": true, "checksum": "7caf1c0ea8b372083328f72f55b3e1e04bf57952", "dest": "/home/giridhara_prasad/default-jiva.j2", "gid": 1000, "group": "giridhara_prasad", "md5sum": "91c01d9718f304c2a324e4a0b623800d", "mode": "0644", "owner": "giridhara_prasad", "size": 608, "src": "/home/giridhara_prasad/.ansible/tmp/ansible-tmp-1534903834.52-25803604662264/source", "state": "file", "uid": 1000}

TASK [k8s-custom-storageclass : template] *************************************************************************************************************************************************************************************************
task path: /home/giridhara_prasad/openebs/e2e/ansible/roles/k8s-custom-storageclass/tasks/main.yml:14
changed: [kubemaster01] => {"changed": true, "checksum": "b07b4aae2f3f22660fb69dde70cf4935d86931c0", "dest": "/home/giridhara_prasad/default-cstor.j2", "gid": 1000, "group": "giridhara_prasad", "md5sum": "681d35b35e0f664da5dcc9842e3a9404", "mode": "0644", "owner": "giridhara_prasad", "size": 467, "src": "/home/giridhara_prasad/.ansible/tmp/ansible-tmp-1534903835.3-152867846298963/source", "state": "file", "uid": 1000}

TASK [k8s-custom-storageclass : template] *************************************************************************************************************************************************************************************************
task path: /home/giridhara_prasad/openebs/e2e/ansible/roles/k8s-custom-storageclass/tasks/main.yml:18
changed: [kubemaster01] => {"changed": true, "checksum": "e92ec255af8c908546b116d73ff415c132fd615d", "dest": "/home/giridhara_prasad/default-cstor-sparse.j2", "gid": 1000, "group": "giridhara_prasad", "md5sum": "52ad7e8d668466279c4c48d291df23cc", "mode": "0644", "owner": "giridhara_prasad", "size": 495, "src": "/home/giridhara_prasad/.ansible/tmp/ansible-tmp-1534903835.78-279896994950605/source", "state": "file", "uid": 1000}

TASK [k8s-custom-storageclass : Delete if the jiva StorageClass Already Exists] ***********************************************************************************************************************************************************
task path: /home/giridhara_prasad/openebs/e2e/ansible/roles/k8s-custom-storageclass/tasks/main.yml:22
fatal: [kubemaster01]: FAILED! => {"changed": true, "cmd": "source ~/.profile; kubectl delete -f /home/giridhara_prasad/default-jiva.j2", "delta": "0:00:00.909359", "end": "2018-08-22 07:40:37.318264", "msg": "non-zero return code", "rc": 1, "start": "2018-08-22 07:40:36.408905", "stderr": "Error from server (NotFound): error when deleting \"/home/giridhara_prasad/default-jiva.j2\": storageclasses.storage.k8s.io \"openebs-jiva-e2e\" not found", "stderr_lines": ["Error from server (NotFound): error when deleting \"/home/giridhara_prasad/default-jiva.j2\": storageclasses.storage.k8s.io \"openebs-jiva-e2e\" not found"], "stdout": "", "stdout_lines": []}
...ignoring

TASK [k8s-custom-storageclass : Create storage class for jiva storage engine.] ************************************************************************************************************************************************************
task path: /home/giridhara_prasad/openebs/e2e/ansible/roles/k8s-custom-storageclass/tasks/main.yml:28
changed: [kubemaster01] => {"attempts": 1, "changed": true, "cmd": "source ~/.profile; kubectl apply -f /home/giridhara_prasad/default-jiva.j2", "delta": "0:00:01.628721", "end": "2018-08-22 07:40:39.166506", "rc": 0, "start": "2018-08-22 07:40:37.537785", "stderr": "", "stderr_lines": [], "stdout": "storageclass \"openebs-jiva-e2e\" created", "stdout_lines": ["storageclass \"openebs-jiva-e2e\" created"]}

TASK [k8s-custom-storageclass : Delete if the cstor StorageClass Already Exists] **********************************************************************************************************************************************************
task path: /home/giridhara_prasad/openebs/e2e/ansible/roles/k8s-custom-storageclass/tasks/main.yml:37
fatal: [kubemaster01]: FAILED! => {"changed": true, "cmd": "source ~/.profile; kubectl delete -f /home/giridhara_prasad/default-cstor.j2", "delta": "0:00:00.896537", "end": "2018-08-22 07:40:40.285794", "msg": "non-zero return code", "rc": 1, "start": "2018-08-22 07:40:39.389257", "stderr": "Error from server (NotFound): error when deleting \"/home/giridhara_prasad/default-cstor.j2\": storageclasses.storage.k8s.io \"openebs-cstor-e2e\" not found", "stderr_lines": ["Error from server (NotFound): error when deleting \"/home/giridhara_prasad/default-cstor.j2\": storageclasses.storage.k8s.io \"openebs-cstor-e2e\" not found"], "stdout": "", "stdout_lines": []}
...ignoring

TASK [k8s-custom-storageclass : Create a storage class for cStor engine] ******************************************************************************************************************************************************************
task path: /home/giridhara_prasad/openebs/e2e/ansible/roles/k8s-custom-storageclass/tasks/main.yml:43
changed: [kubemaster01] => {"attempts": 1, "changed": true, "cmd": "source ~/.profile; kubectl apply -f /home/giridhara_prasad/default-cstor.j2", "delta": "0:00:01.632458", "end": "2018-08-22 07:40:42.132014", "rc": 0, "start": "2018-08-22 07:40:40.499556", "stderr": "", "stderr_lines": [], "stdout": "storageclass \"openebs-cstor-e2e\" created", "stdout_lines": ["storageclass \"openebs-cstor-e2e\" created"]}

TASK [k8s-custom-storageclass : Delete if the cstor StorageClass using sparse file Already Exists] ****************************************************************************************************************************************
task path: /home/giridhara_prasad/openebs/e2e/ansible/roles/k8s-custom-storageclass/tasks/main.yml:52
fatal: [kubemaster01]: FAILED! => {"changed": true, "cmd": "source ~/.profile; kubectl delete -f /home/giridhara_prasad/default-cstor-sparse.j2", "delta": "0:00:01.037796", "end": "2018-08-22 07:40:43.368746", "msg": "non-zero return code", "rc": 1, "start": "2018-08-22 07:40:42.330950", "stderr": "Error from server (NotFound): error when deleting \"/home/giridhara_prasad/default-cstor-sparse.j2\": storageclasses.storage.k8s.io \"openebs-cstor-default-sparse-0.7.0\" not found", "stderr_lines": ["Error from server (NotFound): error when deleting \"/home/giridhara_prasad/default-cstor-sparse.j2\": storageclasses.storage.k8s.io \"openebs-cstor-default-sparse-0.7.0\" not found"], "stdout": "", "stdout_lines": []}
...ignoring

TASK [k8s-custom-storageclass : Create a storage class for cStor engine using sparse file] ************************************************************************************************************************************************
task path: /home/giridhara_prasad/openebs/e2e/ansible/roles/k8s-custom-storageclass/tasks/main.yml:58
changed: [kubemaster01] => {"attempts": 1, "changed": true, "cmd": "source ~/.profile; kubectl apply -f /home/giridhara_prasad/default-cstor-sparse.j2", "delta": "0:00:01.756932", "end": "2018-08-22 07:40:45.327652", "rc": 0, "start": "2018-08-22 07:40:43.570720", "stderr": "", "stderr_lines": [], "stdout": "storageclass \"openebs-cstor-default-sparse-0.7.0\" created", "stdout_lines": ["storageclass \"openebs-cstor-default-sparse-0.7.0\" created"]}

TASK [k8s-custom-storageclass : Remove the copied templates] ******************************************************************************************************************************************************************************
task path: /home/giridhara_prasad/openebs/e2e/ansible/roles/k8s-custom-storageclass/tasks/main.yml:67
changed: [kubemaster01] => (item=default-jiva.j2) => {"changed": true, "item": "default-jiva.j2", "path": "/home/giridhara_prasad/default-jiva.j2", "state": "absent"}
changed: [kubemaster01] => (item=default-cstor.j2) => {"changed": true, "item": "default-cstor.j2", "path": "/home/giridhara_prasad/default-cstor.j2", "state": "absent"}
changed: [kubemaster01] => (item=default-cstor-sparse.j2) => {"changed": true, "item": "default-cstor-sparse.j2", "path": "/home/giridhara_prasad/default-cstor-sparse.j2", "state": "absent"}
META: ran handlers
META: ran handlers

PLAY RECAP ********************************************************************************************************************************************************************************************************************************
kubemaster01               : ok=13   changed=11   unreachable=0    failed=0
```


```
giridhara_prasad@strong-eon-153112:~/openebs/e2e/ansible$ kubectl get sc
NAME                                 PROVISIONER                                                AGE
openebs-cstor-default-sparse-0.7.0   openebs.io/provisioner-iscsi                               1m
openebs-cstor-e2e                    openebs.io/provisioner-iscsi                               1m
openebs-jiva-e2e                     openebs.io/provisioner-iscsi                               1m
openebs-snapshot-promoter            volumesnapshot.external-storage.k8s.io/snapshot-promoter   7d
openebs-standard                     openebs.io/provisioner-iscsi                               1d
```
